File: python/trab1/magnitude.py
# coding=utf-8
from math import log, floor

z = 1.0
while z > 0:
    m = z
    z = z / 2
n = floor(log(abs(log(m, 2) + 1), 2))
print 'Magnitude: %g' % m
print 'Número de bits do expoente: %i' % n

File: ruby/trab1/magnitude.rb
# encoding: utf-8
include Math

z = 1.0
while z > 0.0 do
  m = z
  z /= 2.0
end
n = (log((log(m) / log(2.0)).abs + 1.0) / log(2.0)).floor
puts "Magnitude: #{m}"
puts "Número de bits do expoente: #{n}"

